src/flow.jl

```diff
@@ -452,42 +455,21 @@ function update(ssf::LateralSSF, network, frac_toriver, ksat_profile)
                     upstream_nodes[n],
                     eltype(ssf.to_river),
                 )
-                if (ksat_profile == "exponential") ||
-                   (ksat_profile == "exponential_constant")
-                    ssf.ssf[v], ssf.zi[v], ssf.exfiltwater[v] = kinematic_wave_ssf(
-                        ssf.ssfin[v],
-                        ssf.ssf[v],
-                        ssf.zi[v],
-                        ssf.recharge[v],
-                        ssf.kh_0[v],
-                        ssf.slope[v],
-                        ssf.theta_s[v] - ssf.theta_r[v],
-                        ssf.f[v],
-                        ssf.soilthickness[v],
-                        ssf.dt,
-                        ssf.dl[v],
-                        ssf.dw[v],
-                        ssf.ssfmax[v],
-                        ssf.z_exp[v],
-                        ksat_profile,
-                    )
-                elseif (ksat_profile == "layered") ||
-                       (ksat_profile == "layered_exponential")
-                    ssf.ssf[v], ssf.zi[v], ssf.exfiltwater[v] = kinematic_wave_ssf(
-                        ssf.ssfin[v],
-                        ssf.ssf[v],
-                        ssf.zi[v],
-                        ssf.recharge[v],
-                        ssf.kh[v],
-                        ssf.slope[v],
-                        ssf.theta_s[v] - ssf.theta_r[v],
-                        ssf.soilthickness[v],
-                        ssf.dt,
-                        ssf.dl[v],
-                        ssf.dw[v],
-                        ssf.ssfmax[v],
-                    )
-                end
+                ssf.ssf[v], ssf.zi[v], ssf.exfiltwater[v] = kinematic_wave_ssf(
+                    ssf.ssfin[v],
+                    ssf.ssf[v],
+                    ssf.zi[v],
+                    ssf.recharge[v],
+                    ssf.slope[v],
+                    ssf.theta_s[v] - ssf.theta_r[v],
+                    ssf.soilthickness[v],
+                    ssf.dt,
+                    ssf.dl[v],
+                    ssf.dw[v],
+                    ssf.ssfmax[v],
+                    ssf.kh_profile,
```

Comment:
Bit of background: we decided to implement vertical and horizontal process functions (for each grid cell) at a low level (with scalar arguments). Because of using the `kh_profile` here for dispatching we deviate a bit from this approach. As this struct will change for v1 (also using `variables`, `parameters` and `boundary_conditions` (if applicable)) I suggest we come back to this as part of refactoring the Wflow routing part.